### PR TITLE
Screenshots paths made more user-friendly (closes #650)

### LINKED ISSUE #650: Make screenshots paths more user-friendly
We've discussed this with @superroma and here is the idea how we can transform screenshot paths:
`YYYY-MM-DD_HH-MM-SS/{user-agent}/{1,2,3,...}.jpg`

For errors:
`YYYY-MM-DD_HH-MM-SS/{user-agent}/errors/{1,2,3,...}.jpg`

Date is the start time of the test task. Screenshots are sequentially numbered within directories.
BTW, currently we manually escape user agent in path, I suggest to use https://www.npmjs.com/package/sanitize-filename package for this purpose.

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin @AlexanderMoskovkin @VasilyStrelyaev

### CODE REVIEW

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by VasilyStrelyaev** [comment]
lgtm

File: package.json [hunk 1]
```diff
@@ -59,7 +59,8 @@
     "testcafe-reporter-spec": "^2.0.0",
     "testcafe-reporter-xunit": "^2.0.0",
     "time-limit-promise": "^1.0.2",
-    "useragent": "^2.1.7"
+    "useragent": "^2.1.7",
+    "dateformat": "^1.0.12"
```
Comment (inikulin) [inline package.json]: We have a `moment` for this

File: src/runner/browser-job.js [hunk 2]
```diff
@@ -83,12 +83,12 @@ export default class BrowserJob extends EventEmitter {
             this.emit('done');
     }
 
-    _createTestRun (test) {
+    _createTestRun (test, testIndex, quarantineAttempt) {
```
Comment (inikulin) [inline src/runner/browser-job.js]: `quarantineAttemptNum`

File: src/screenshots/capturer.js [hunk 3]
```diff
@@ -1,66 +1,112 @@
...
+        this.testDirName          = Capturer._getTestDirName(namingOptions.testIndex, namingOptions.quarantineAttempt);
+        this.screenshotIndex      = 1;
+        this.errorScreenshotIndex = 1;
+        this.pathCustomized       = false;
     }
 
-    static _getFileName (stepName) {
-        return `${stepName && stepName.replace(/\s|\\|\/|"|\*|\?|<|>|\|/g, '_') || 'page-load'}.png`;
+    static _correctFilePath (path) {
+        var correctedPath = path
```
Comment (inikulin) [inline src/screenshots/capturer.js]: `sanitizedPath`

and do we need to perform split here? I thought `sanitize` can handle it

File: src/screenshots/capturer.js [hunk 4]
```diff
@@ -1,66 +1,112 @@
 import { join as joinPath, dirname } from 'path';
 import promisify from '../utils/promisify';
+import sanitize from 'sanitize-filename';
```
Comment (inikulin) [inline src/screenshots/capturer.js]: `sanitizeFilename`

File: src/screenshots/index.js [hunk 5]
```diff
@@ -1,30 +1,58 @@
...
 export default class Screenshots {
     constructor (path) {
         this.enabled         = !!path;
         this.screenshotsPath = path;
         this.testEntries     = [];
+        this.startDate       = Screenshots._getStartTestDate();
+        this.userAgentNames  = [];
+    }
+
+    static _getStartTestDate () {
```
Comment (inikulin) [inline src/screenshots/index.js]: `_getTaskStartDate`

Comment (inikulin) [inline src/screenshots/index.js]: Also it's not quite correct (and `startDate` var name as well) - it's not a date object, it's formatted date and more likely it's semantically `baseDir` or something like that.

File: src/screenshots/capturer.js (hunk 3 above)
Comment (helen-dikareva) [inline src/screenshots/capturer.js]: Because `sanitize` removes `/` symbols

**Comment by inikulin** [comment]
\r-

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

**Comment by inikulin** [comment]
Why we still have date format as dependency?

**Comment by helen-dikareva** [comment]
no reason

**Comment by helen-dikareva** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
FYI: it's a breaking change, so we will need to update version to 0.3.x-alpha

**Comment by inikulin** [comment]
@AlexanderMoskovkin Why it's a breaknig change? Is any public API affected?

**Comment by AlexanderMoskovkin** [comment]
Public API is not changed but screenshots dirs are changed. For example if @kirovboris updates the `testcafe` version in the `testcafe-studio` project the tests are failed because screenshots check

**Comment by inikulin** [comment]
Please, don't envolve your feelings into versioning - it's a common mistake. We have a well-defined protocol for this. (BTW you all right that this is a major version bump, but you're absolutely wrong why it's a major version bump, it bothers me)

**Comment by inikulin** [comment]
Tip: google definition of API

**Pull request merged**